src/App.jsx

```diff
@@ -14,33 +20,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, sortField, query) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (query) {
+    preparedGoods.reverse();
+  }
```

Comment:
The logic here is incorrect. The `query` parameter is being used to reverse the goods, which is misleading. You should use the `reversed` state instead to determine if the goods should be reversed. Update the function to use `reversed` instead of `query`.